src/components/dialogs/network-modifications/by-filter/by-assignment/assignment/assignment-form.tsx

```diff
@@ -0,0 +1,158 @@
+    }, [watchPropertyName, predefinedProperties]);
+
+    const options = useMemo(() => {
+        return equipmentFields?.find((fieldOption) => fieldOption?.id === watchEditedField)?.values ?? [];
+    }, [watchEditedField, equipmentFields]);
+
+    // reset value field only when data type is changed
+    const prevDataType = usePrevious(dataType);
+    // important, check prevDataType should not be undefined to ensure that setValue is not called in initialization
+    if (prevDataType && prevDataType !== dataType) {
+        setValue(`${name}.${index}.${VALUE_FIELD}`, dataType === DataType.BOOLEAN ? false : null);
+    }
+
+    const filtersField = (
+        <DirectoryItemsInput
+            name={`${name}.${index}.${FILTERS}`}
+            equipmentTypes={[equipmentType]}
+            elementType={ElementType.FILTER}
+            label={'filter'}
+            titleId={'FiltersListsSelection'}
+            disable={!equipmentType}
+        />
+    );
+
+    const editedField = (
+        <AutocompleteInput
+            name={`${name}.${index}.${EDITED_FIELD}`}
+            options={equipmentFields}
+            label={'EditedField'}
+            size={'small'}
+            inputTransform={(value: any) => equipmentFields.find((option) => option?.id === value) || value}
+            outputTransform={(option: any) => getIdOrValue(option) ?? null}
+            getOptionLabel={(option: any) => (option ? intl.formatMessage({ id: getLabelOrValue(option) }) : option)}
+            isOptionEqualToValue={areIdsEqual}
+        />
+    );
+
+    const propertyNameField = (
+        <AutocompleteInput
+            name={`${name}.${index}.${PROPERTY_NAME_FIELD}`}
+            options={predefinedPropertiesNames}
+            label={'PropertyName'}
+            size={'small'}
+            allowNewValue
+        />
+    );
+
+    const valueField = useMemo(() => {
+        if (dataType === DataType.PROPERTY) {
+            return (
+                <AutocompleteInput
+                    name={`${name}.${index}.${VALUE_FIELD}`}
+                    label={'PropertyValue'}
+                    options={predefinedPropertiesValues}
+                    size={'small'}
+                    allowNewValue
+                />
+            );
+        }
+
+        if (dataType === DataType.INTEGER) {
+            return <IntegerInput name={`${name}.${index}.${VALUE_FIELD}`} label={'Value'} />;
+        }
+
+        if (dataType === DataType.BOOLEAN) {
+            return <SwitchInput name={`${name}.${index}.${VALUE_FIELD}`} formProps={{ value: false }} />;
+        }
+
+        if (dataType === DataType.ENUM) {
+            return (
+                <SelectInput name={`${name}.${index}.${VALUE_FIELD}`} label="Value" options={options} size={'small'} />
+            );
+        }
+
+        // by default is a numeric type
+        return <FloatInput name={`${name}.${index}.${VALUE_FIELD}`} label={'Value'} />;
```

Comment:
Why do you sometimes write `label="Value"` and sometimes `label={'Value'}`. Does it make a difference ?

Reply:
DONE

Reply:
DONE